frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -114,31 +114,65 @@ export default class FirebaseInteractor {
    *
    * @returns { Promise<Assessment> }
    */
-  async getAssessment(): Promise<Assessment> {
-    let assessmentRef = await this.db.collection("assessments").get();
-    let assessmentDocs = assessmentRef.docs;
+  async getAssessment(firebaseId: string): Promise<Assessment> {
+    let assessmentRef = await this.db.collection("assessments").doc(firebaseId).get();
     // Hardcoding first assessment for now
-    let assessment = assessmentDocs[0];
-    let { id, currentIndex } = assessment.data();
-    let wordPath = `assessments/${assessment.id}/words`;
-    let wordRef = await this.db.collection(wordPath).get();
-    let wordDocs = wordRef.docs;
-
-    let words: Word[] = [];
-    for (let wordRef of wordDocs) {
+    let assessment = assessmentRef.data();
+    if (assessment == null) {
+      throw new Error(`Assessment with id ${firebaseId} does not exist`)
+    }
+    let { id, currentIndex, words } = assessment;
+
+    let actualWords: Word[] = [];
+    for (let wordString of words) {
+      let wordRef = await this.db.collection("words").doc(wordString).get();
       let word = wordRef.data();
-      words.push({
+      if (word == null) {
+        throw new Error(`There is no word with the id ${wordString}`);
+      }
+      actualWords.push({
         value: word.value,
         correctImage: word.correctImage,
         incorrectImages: word.incorrectImages,
         id: wordRef.id,
         createdAt: word.dateCreated.toDate(),
       });
     }
-    words.sort(
+    actualWords.sort(
       (word1, word2) => word1.createdAt.getTime() - word2.createdAt.getTime()
     );
-    return { id, currentIndex, words };
+    return { id, currentIndex, words: actualWords, firebaseId: assessment.id };
+  }
+
+  async updateAssessment(id: string, responses: AssessmentResult[]) {
+    await Promise.all(responses.map(async (response) => {
+      return await this.db.collection("assessments").doc(id).collection("results").doc(response.word).set({correct: response.correct});
+    }))
+  }
+
+  async createInterventionSet(id: string, responses: AssessmentResult[]) {
+    await Promise.all(responses.map(async (response) => {
+
+      let word = (await this.db.collection("words").doc(response.word));
```

Comment:
I wonder if we should make a helper function that given an id, gets the corresponding word?